### ISSUE TITLE
Ansible options overwrite with Linchpin API

### ISSUE THREAD

**Issue by abraverm** [CentOS-PaaS-SIG/linchpin#1125]
[Ansible configurations are currently hard coded](https://github.com/CentOS-PaaS-SIG/linchpin/blob/696d7a08b6a58c1b147432203fddc6d5c55ce0e7/linchpin/ansible_runner.py#L122-L162) and the only way to overwrite them is with `ansible.cfg` file. Linchpin API should allow overwrite Ansible configurations without using `ansible.cfg` but programmatically. This is needed for custom configurations such as `log_path`.

**Comment by samvarankashyap** [comment]
part of #1057

**Comment by samvarankashyap** [comment]
Shall be worked on implemented in 2.0 sprint.

**Comment by samvarankashyap** [comment]
@abraverm after brief investigation it is found that ansible.cfg parameters which are not mentioned inside the [options](https://github.com/CentOS-PaaS-SIG/linchpin/blob/696d7a08b6a58c1b147432203fddc6d5c55ce0e7/linchpin/ansible_runner.py#L122-L162)
Such as log_path cannot be managed through ansible API. 
However, a workaround can be generating ansible.cfg before provisioning workspace.
 
Given the efforts, am not sure it would be worthwhile or not. 
@abraverm @14rcole  any thoughts around this  ???

**Comment by 14rcole** [comment]
Can we just provide an option in linchpin.conf `ansible_config_path` that provides a path to an `ansible.cfg`?  That way users can copy it into their workspace and modify it to their liking

**Comment by abraverm** [comment]
How about using environment variables (inside Linchpin process)?
https://docs.ansible.com/ansible/latest/reference_appendices/config.html#envvar-ANSIBLE_LOG_PATH

**Comment by 14rcole** [comment]
@abraverm if that supports all of the options you could set in `ansible.cfg` I don't have a problem with it

**Comment by abraverm** [comment]
I think so...

**Comment by samvarankashyap** [comment]
Using environment variables seems like a better idea. 
I will try to add an option to linchpin command line to add environment vars dynamically to ansible_runner.
That can help us harnessing all types of environment variables for linchpin.

**Comment by samvarankashyap** [comment]
@abraverm @14rcole 
I have implemented the part passing environment variables to Ansible playbooks. 
However, In my testing I found out ansible does not allow you to override ANSIBLE configuration parameters even though we pass it as environment variables. 
This is due to the fact that ansible reads its environment variables from system processes rather than the python process. 
However, We will be able to override all the option variables mentioned in 
https://github.com/CentOS-PaaS-SIG/linchpin/blob/696d7a08b6a58c1b147432203fddc6d5c55ce0e7/linchpin/ansible_runner.py#L122 
With keyword arguments. 
I will be soon creating a PR for it. 
Other than those options we cannot override ansible of options. 

Alternative:
The above functionality can be achieved by replacing the ansible_runner.py with subprocess implementation. Which is very difficult at this point since ansible is not allowing nested dictionaries( ie., in our case topology and layout, resources) to be passed as extra-vars. 
We need to investigate into other alternatives.
